src/main/java/org/opensearch/timeseries/indices/IndexManagement.java

```diff
@@ -88,308 +72,698 @@
+    }
+
+    protected void deleteIndexIteration(String[] toDelete) {
+        for (String index : toDelete) {
+            DeleteIndexRequest singleDeleteRequest = new DeleteIndexRequest(index);
+            adminClient.indices().delete(singleDeleteRequest, ActionListener.wrap(singleDeleteResponse -> {
+                if (!singleDeleteResponse.isAcknowledged()) {
+                    logger.error("Retrying deleting {} does not succeed.", index);
+                }
+            }, exception -> {
+                if (exception instanceof IndexNotFoundException) {
+                    logger.info("{} was already deleted.", index);
+                } else {
+                    logger.error(new ParameterizedMessage("Retrying deleting {} does not succeed.", index), exception);
+                }
+            }));
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    protected void shouldUpdateConcreteIndex(String concreteIndex, Integer newVersion, ActionListener<Boolean> thenDo) {
+        IndexMetadata indexMeataData = clusterService.state().getMetadata().indices().get(concreteIndex);
+        if (indexMeataData == null) {
+            thenDo.onResponse(Boolean.FALSE);
             return;
         }
-        String resultMapping = getAnomalyResultMappings();
+        Integer oldVersion = CommonValue.NO_SCHEMA_VERSION;
 
-        Map<String, Object> asMap = XContentHelper.convertToMap(new BytesArray(resultMapping), false, XContentType.JSON).v2();
-        Object properties = asMap.get(CommonName.PROPERTIES);
-        if (properties instanceof Map) {
-            AD_RESULT_FIELD_CONFIGS = (Map<String, Object>) properties;
-        } else {
-            logger.error("Fail to read result mapping file.");
+        Map<String, Object> indexMapping = indexMeataData.mapping().getSourceAsMap();
+        Object meta = indexMapping.get(IndexManagement.META);
+        if (meta != null && meta instanceof Map) {
+            Map<String, Object> metaMapping = (Map<String, Object>) meta;
+            Object schemaVersion = metaMapping.get(org.opensearch.timeseries.constant.CommonName.SCHEMA_VERSION_FIELD);
+            if (schemaVersion instanceof Integer) {
+                oldVersion = (Integer) schemaVersion;
+            }
         }
+        thenDo.onResponse(newVersion > oldVersion);
     }
 
-    /**
-     * Get anomaly detector index mapping json content.
-     *
-     * @return anomaly detector index mapping
-     * @throws IOException IOException if mapping file can't be read correctly
-     */
-    public static String getAnomalyDetectorMappings() throws IOException {
-        URL url = AnomalyDetectionIndices.class.getClassLoader().getResource(INDEX_MAPPING_FILE);
-        return Resources.toString(url, Charsets.UTF_8);
+    protected void updateJobIndexSettingIfNecessary(String indexName, IndexState jobIndexState, ActionListener<Void> listener) {
+        GetSettingsRequest getSettingsRequest = new GetSettingsRequest()
```

Comment:
was this functionality there in some other block of the AD plugin code?

Reply:
ignore previous comment - I see the deleted lines. I think the patch tool got confused ..